README.md
# Venipak COD for Prestashop 1.6, 1.7, 8.x
Venipak COD module for Prestashop 1.6.0 - 8.1.x

### Minimum requirements
- Prestashop 1.6.0
- PHP 7.0

## Installing the module

Download the venipakcod.zip from the link below:

[![Download](https://img.shields.io/badge/dynamic/json.svg?label=download&url=https://api.github.com/repos/venipak/prestashop-venipak-cod-1.6-1.7/releases/latest&query=$.assets[0].name&style=for-the-badge)](https://github.com/venipak/prestashop-venipak-cod-1.6-1.7/releases/latest/download/venipakcod.zip)

Follow the installation instructions provided here: https://www.venipak.com/wp-content/uploads/Venipak-Prestashop-User-Manualm.pdf


CHANGELOG.md
# Changelog

## [1.0.4] - 2023-09-06
### Improved
- added Changelog file
- the module is adapted to work in Prestashop system versions 8.0 and 8.1


translations/en.php
<?php

global $_MODULE;
$_MODULE = array();

$_MODULE['<{cashondelivery}prestashop>validation_ea9cf7e47ff33b2be14e6dd07cbcefc6'] = 'Shipping';
$_MODULE['<{cashondelivery}prestashop>validation_f1d3b424cd68795ecaa552883759aceb'] = 'Order summary';
$_MODULE['<{cashondelivery}prestashop>validation_d538c5b86e9a71455ba27412f4e9ab51'] = 'Venipak Cash on Delivery (COD) payment';
$_MODULE['<{cashondelivery}prestashop>validation_ad2f6328c24caa7d25dd34bfc3e7e25a'] = 'You have chosen the Cash on Delivery method.';
$_MODULE['<{cashondelivery}prestashop>validation_e2867a925cba382f1436d1834bb52a1c'] = 'The total amount of your order is';
$_MODULE['<{cashondelivery}prestashop>validation_1f87346a16cf80c372065de3c54c86d9'] = '(tax incl.)';
$_MODULE['<{cashondelivery}prestashop>validation_52f64bc0164b0e79deaeaaaa7e93f98f'] = 'Please confirm your order by clicking \'I confirm my order\'.';
$_MODULE['<{cashondelivery}prestashop>validation_569fd05bdafa1712c4f6be5b153b8418'] = 'Other payment methods';
$_MODULE['<{cashondelivery}prestashop>validation_46b9e3665f187c739c55983f757ccda0'] = 'I confirm my order';
$_MODULE['<{cashondelivery}prestashop>confirmation_2e2117b7c81aa9ea6931641ea2c6499f'] = 'Your order on';
$_MODULE['<{cashondelivery}prestashop>confirmation_75fbf512d744977d62599cc3f0ae2bb4'] = 'is complete.';
$_MODULE['<{cashondelivery}prestashop>confirmation_8861c5d3fa54b330d1f60ba50fcc4aab'] = 'You have chosen the cash on delivery method.';
$_MODULE['<{cashondelivery}prestashop>confirmation_e6dc7945b557a1cd949bea92dd58963e'] = 'Your order will be sent very soon.';
$_MODULE['<{cashondelivery}prestashop>confirmation_0db71da7150c27142eef9d22b843b4a9'] = 'For any questions or for further information, please contact our';
$_MODULE['<{cashondelivery}prestashop>confirmation_64430ad2835be8ad60c59e7d44e4b0b1'] = 'customer support';
$_MODULE['<{cashondelivery}prestashop>payment_b7ada96a0da7ee7fb5371cca0b036d5c'] = 'Pay with Venipak cash on delivery (COD)';
$_MODULE['<{cashondelivery}prestashop>payment_536dc7424180872c8c2488ae0286fb53'] = 'You pay for the merchandise upon delivery';


return $_MODULE;


translations/lt.php
<?php

global $_MODULE;
$_MODULE = array();
$_MODULE['<{venipakcod}prestashop>venipakcod_c888438d14855d7d96a2724ee9c306bd'] = 'Nustatymai atnaujinti';
$_MODULE['<{venipakcod}prestashop>venipakcod_310cdcd76c50cb7ee53563ddfa2a6cf5'] = 'Mokestis:';
$_MODULE['<{venipakcod}prestashop>venipakcod_590525891b28a45bb62edafdbc7aa773'] = 'Naudoti % mokestį?';
$_MODULE['<{venipakcod}prestashop>venipakcod_7469a286259799e5b37e5db9296f00b3'] = 'TAIP';
$_MODULE['<{venipakcod}prestashop>venipakcod_c2f3f489a00553e7a01d369c103c7251'] = 'NE';
$_MODULE['<{venipakcod}prestashop>venipakcod_2d637642e4a51dbe4c0851c3fd518362'] = 'Fiksuotas mokestis';
$_MODULE['<{venipakcod}prestashop>venipakcod_2d11bd51ce1500c0896e62ad6c47aa88'] = '% mokestis';
$_MODULE['<{venipakcod}prestashop>venipakcod_c1ef7a9ff2e232239ecaf3720b97db2c'] = 'Užsakymo suma';
$_MODULE['<{venipakcod}prestashop>venipakcod_ce31e2a082d17e038fcc6e3006166653'] = 'Min';
$_MODULE['<{venipakcod}prestashop>venipakcod_26a4b44a837bf97b972628509912b4a5'] = 'Maks';
$_MODULE['<{venipakcod}prestashop>venipakcod_77e64b0d0b130bb54c89de3236488d9e'] = 'Min. Maks. nustato rėžį kuriam bus taikomas mokestis';
$_MODULE['<{venipakcod}prestashop>venipakcod_5ead431e0cbdaf1d1436c22c58cbc173'] = 'Siuntimo būdai';
$_MODULE['<{venipakcod}prestashop>venipakcod_c9cc8cce247e49bae79f15173ce97354'] = 'Išsaugoti';
$_MODULE['<{venipakcod}prestashop>venipakcod_630f6dc397fe74e52d5189e2c80f282b'] = 'Atgal';
$_MODULE['<{venipakcod}prestashop>venipakcod_27d6e93ae9563c785c8abf7b416577a1'] = 'Pasirinkite atsiėmimo punktą';
$_MODULE['<{venipakcod}prestashop>venipakcod_ceb33e523816be5c557687f9e7fb57be'] = 'Pristatant siuntą galėsite atsiskaityti banko kortele.';
$_MODULE['<{venipakcod}prestashop>venipakcod_4193b2ca45352de61ba304aa78673e64'] = 'Šis mokėjimo būdas prie siuntimo kainos pridės papildomą mokestį (%s).';
$_MODULE['<{venipakcod}prestashop>venipakcod_25b1de072279d6df1f776c50e6cdefee'] = 'Mokėti pristatymo metu (COD)';
$_MODULE['<{venipakcod}prestashop>venipakcod_b7ada96a0da7ee7fb5371cca0b036d5c'] = 'Sumokėti pristatymo metu (COD)';
$_MODULE['<{venipakcod}prestashop>validation_644818852b4dd8cf9da73543e30f045a'] = 'Grįžti į atsiskaitymą';
$_MODULE['<{venipakcod}prestashop>validation_6ff063fbc860a79759a7369ac32cee22'] = 'Atsiskaitymas';
$_MODULE['<{venipakcod}prestashop>validation_f1d3b424cd68795ecaa552883759aceb'] = 'Užsakymo santrauka';
$_MODULE['<{venipakcod}prestashop>validation_e2867a925cba382f1436d1834bb52a1c'] = 'Užsakymo viso: ';
$_MODULE['<{venipakcod}prestashop>validation_1f87346a16cf80c372065de3c54c86d9'] = '(įskaitant mok.)';
$_MODULE['<{venipakcod}prestashop>validation_52f64bc0164b0e79deaeaaaa7e93f98f'] = 'Prašome patvirtinti užsakymą paspaudžiant mygtuką \'Patvirtinti užsakymą\'';
$_MODULE['<{venipakcod}prestashop>validation_569fd05bdafa1712c4f6be5b153b8418'] = 'Kiti atsiskaitymo būdai';
$_MODULE['<{venipakcod}prestashop>validation_46b9e3665f187c739c55983f757ccda0'] = 'Patvirtinti užsakymą';
$_MODULE['<{venipakcod}prestashop>confirmation_88526efe38fd18179a127024aba8c1d7'] = 'Jūsų užaskymas %s patvirtintas ';
$_MODULE['<{venipakcod}prestashop>confirmation_e6dc7945b557a1cd949bea92dd58963e'] = 'Užsakymas bus išsiųstas greitu metu.';
$_MODULE['<{venipakcod}prestashop>confirmation_0db71da7150c27142eef9d22b843b4a9'] = 'Kilus klausimams prašome kreiptis į mūsų';
$_MODULE['<{venipakcod}prestashop>confirmation_64430ad2835be8ad60c59e7d44e4b0b1'] = 'klientų aptarnavimą';
$_MODULE['<{venipakcod}prestashop>payment_d496fe292405ff53fbcf49386fc269f4'] = 'COD mokestis ';
$_MODULE['<{venipakcod}prestashop>payment_0c9778c2abf1ab2b306ac9ee24122b02'] = 'Patvirtinus užsakymą COD mokestis bus pridėtas prie pristatymo mokesčio';
$_MODULE['<{venipakcod}prestashop>payment_536dc7424180872c8c2488ae0286fb53'] = 'Už prekes atsiskaitysite pristatymo metu';
$_MODULE['<{venipakcod}prestashop>ajax_818e3f6aee02ea764a6a411f063e1bb7'] = 'Jūsų pasirinktas atsiėmimo punktas nepalaiko C.O.D mokėjimo būdo. Pasirinkite kitą atsiėmimo punktą';


controllers/front/ajax.php
<?php
class VenipakCodAjaxModuleFrontController extends ModuleFrontController
{
    /**
     * @see FrontController::initContent()
     */
    public function initContent()
    {
        parent::initContent();
        $venipakModule = Module::getInstanceByName('mijoravenipak');
        if($venipakModule && $venipakModule->active)
        {
            $id_carrier = $this->context->cart->id_carrier;
            $carrier = new Carrier($id_carrier);
            $carrier_reference = $carrier->id_reference;

            // Check if the selected terminal is C.O.D. Otherwise, prompt user to select another one.
            $terminal_data = Db::getInstance()->getValue((new DbQuery())
                ->select('terminal_info')
                ->from('mjvp_orders')
                ->where('id_cart = ' . $this->context->cart->id)
            );
            $terminal_data = json_decode($terminal_data, true);
            if($carrier_reference == Configuration::get(MijoraVenipak::$_carriers['pickup']['reference_name']) && (isset($terminal_data['is_cod']) && !$terminal_data['is_cod']))
            {
                $content = $venipakModule->hookDisplayCarrierExtraContent(
                    [
                        'cart' => $this->context->cart,
                        'carrier' => (array) $carrier,
                        'filters' => ['cod_enabled' => 1],
                        'venipakcod' => 1
                    ]
                );
                $this->context->smarty->assign(
                    array(
                        'images_url' => $this->module->_path . 'views/images/',
                    )
                );
                die(json_encode([
                    'error' => $this->module->l('Your selected terminal does not support C.O.D payment method. Please select another terminal', 'Ajax'),
                    'carrier_content' => $content,
                    'mjvp_map_template' => $this->context->smarty->fetch(VenipakCod::$_moduleDir . '/views/templates/front/map-template.tpl'),
                ]));
            }
            else
            {
                die(json_encode(['success' => true]));
            }

        }
    }

}


controllers/front/payment.php
<?php

class VenipakCodPaymentModuleFrontController extends ModuleFrontController
{
    public $ssl = true;

    public function postProcess()
    {
        $cart = $this->context->cart;
        if ($cart->id_customer == 0 || $cart->id_address_delivery == 0 || $cart->id_address_invoice == 0 || !$this->module->active)
            Tools::redirectLink(__PS_BASE_URI__ . 'order.php?step=1');

        // Check that this payment option is still available in case the customer changed his address just before the end of the checkout process
        $authorized = false;
        foreach (Module::getPaymentModules() as $module)
            if ($module['name'] == 'venipakcod') {
                $authorized = true;
                break;
            }
        if (!$authorized)
            die($this->module->l('This payment method is not available.'));

        $customer = new Customer($cart->id_customer);
        if (!Validate::isLoadedObject($customer))
            Tools::redirectLink(__PS_BASE_URI__ . 'order.php?step=1');


        $customer = new Customer((int) $cart->id_customer);
        $total = $cart->getOrderTotal(true, Cart::BOTH);
        $this->module->validateOrder((int) $cart->id, Configuration::get('PS_OS_PREPARATION'), $total, $this->module->displayName, null, array(), null, false, $customer->secure_key);
        Tools::redirectLink(__PS_BASE_URI__ . 'order-confirmation.php?key=' . $customer->secure_key . '&id_cart=' . (int) $cart->id . '&id_module=' . (int) $this->module->id . '&id_order=' . (int) $this->module->currentOrder);
    }
}


views/js/terminals_map_init.js
var venipak_custom_modal_cod = function() {
    let mjvp_map_container =  document.getElementById('mjvp-pickup-select-modal');
    let tmjs = null;

    if (typeof(mjvp_map_container) != 'undefined' && mjvp_map_container != null) {
        tmjs = new TerminalMappingMjvp('https://venipak.uat.megodata.com/ws');
        tmjs.setImagesPath(mjvp_imgs_url);
        tmjs.setTranslation(mjvp_terminal_select_translates);

        tmjs.dom.setContainerParent(document.getElementById('mjvp-pickup-select-modal'));
        // tmjs.terminals_cache = null;
        tmjs.init({
            country_code: mjvp_country_code,
            identifier: '',
            isModal: true,
            hideContainer: false,
            hideSelectBtn: false,
            postal_code: mjvp_postal_code,
            city: mjvp_city
        });

        tmjs.sub('tmjs-ready', function(data) {
            let selected_terminal = document.getElementById("mjvp-selected-terminal").value;
            let selected_location = tmjs.map.getLocationById(parseInt(selected_terminal));
            if (typeof(selected_location) != 'undefined' && selected_location != null) {
                tmjs.publish('terminal-selected', selected_location);
                document.querySelector('.tmjs-selected-terminal').innerHTML = '<span class="mjvp-tmjs-terminal-name">' + selected_location.name + '</span> <span class="mjvp-tmjs-terminal-address">(' + selected_location.address + ')</span> <span class="mjvp-tmjs-terminal-comment">' + selected_location.city + '.</span>';
            }
        });
        tmjs.sub('terminal-selected', function(data) {
            document.getElementById("mjvp-selected-terminal").value = data.id;
            mjvp_registerSelectionCod('mjvp-selected-terminal');
            tmjs.publish('close-map-modal');
            document.querySelector('.tmjs-selected-terminal').innerHTML = '<span class="mjvp-tmjs-terminal-name">' + data.name + '</span> <span class="mjvp-tmjs-terminal-address">(' + data.address + ')</span> <span class="mjvp-tmjs-terminal-comment">' + data.city + '.</span>';
        });
    }

    window['venipak_custom_modal_cod'].tmjs = tmjs;
}

controllers/front/validation.php
<?php
class VenipakCodValidationModuleFrontController extends ModuleFrontController
{
    public $ssl = true;
    private $_prefix = 'VENIPAKCOD_';

    public function init()
    {
        $this->display_column_left = false;
        parent::init();
    }

    public function postProcess()
	{
		if ($this->context->cart->id_customer == 0 || $this->context->cart->id_address_delivery == 0 || $this->context->cart->id_address_invoice == 0 || !$this->module->active)
			Tools::redirectLink(__PS_BASE_URI__ . 'order.php?step=1');

		// Check that this payment option is still available in case the customer changed his address just before the end of the checkout process
		$authorized = false;
		foreach (Module::getPaymentModules() as $module)
			if ($module['name'] == 'venipakcod') {
				$authorized = true;
				break;
			}
		if (!$authorized)
			die(Tools::displayError('This payment method is not available.'));

		$customer = new Customer($this->context->cart->id_customer);
		if (!Validate::isLoadedObject($customer))
			Tools::redirectLink(__PS_BASE_URI__ . 'order.php?step=1');

		if (Tools::getValue('confirm')) {
			$customer = new Customer((int) $this->context->cart->id_customer);
			$total = $this->context->cart->getOrderTotal(true, Cart::BOTH);
			$this->module->validateOrder((int) $this->context->cart->id, Configuration::get('PS_OS_PREPARATION'), $total, $this->module->displayName, null, array(), null, false, $customer->secure_key);
			Tools::redirectLink(__PS_BASE_URI__ . 'order-confirmation.php?key=' . $customer->secure_key . '&id_cart=' . (int) $this->context->cart->id . '&id_module=' . (int) $this->module->id . '&id_order=' . (int) $this->module->currentOrder);
		}
	}

	/**
	 * @see FrontController::initContent()
	 */
	public function initContent()
	{
		parent::initContent();

		$this->context->smarty->assign(array(
			'total' => $this->context->cart->getOrderTotal(true, Cart::BOTH),
			'cod_fee' => (float) $this->module->getCodFee($this->context->cart),
			'this_path' => $this->module->getPathUri(), //keep for retro compat
			'this_path_cod' => $this->module->getPathUri(),
			'this_path_ssl' => Tools::getShopDomainSsl(true, true) . __PS_BASE_URI__ . 'modules/' . $this->module->name . '/'
		));

		$this->setTemplate('validation.tpl');
	}
}


views/js/payment-terminal.js
$(document).on('ready', () => {
    // Trying to handle opc disaster...
    if(typeof page_name != "undefined" && page_name == 'order-opc')
    {
        // If TOS is checked, the payment blocks are already visible, so we bind the event on page load..
        if($('.payment_module').length != 0)
        {
            codEventListener();
            filterTerminalsAjax({'cod_enabled' : 1});
        }
        // Otherwise, we bind event only after payments are loaded with Ajax..
        else
        {
            $( document ).ajaxComplete(function( event, xhr, settings ) {
                if ( settings.data.includes('updateTOSStatusAndGetPayments&checked=1')) {
                    codEventListener();
                    // Additionally, run filter to get only COD terminals
                    filterTerminalsAjax({'cod_enabled' : 1});
                }
            });
        }
    }
    else
    {
        codEventListener();
    }
});

function codEventListener() {
    $('.venipakcod').on('click', () => {
        event.preventDefault();
        if(!$(event.target).hasClass('tmjs-open-modal-btn'))
        {
            $('.venipak-service-content').remove();
            $('.venipakcod .alert').remove();
            $(".mjvp-pickup-filter").unbind('click');
            $.ajax({
                type: "POST",
                url: cod_ajax_url,
                dataType: "json",
                success: function (res) {
                    if(typeof res.carrier_content != 'undefined' && typeof res.mjvp_map_template != 'undefined' && res.carrier_content)
                    {
                        var error = '';
                        if(typeof res.error != 'undefined')
                        {
                            error = `<div class="alert alert-danger">${res.error}</div>`;
                        }
                        $('.venipakcod').append(error);
                        mjvp_map_template = res.mjvp_map_template;
                        $('.venipakcod').append(`
                        <div class="venipak-service-content">
                            ${res.carrier_content}
                        </div>`);
                        if($('.tmjs-modal').length != 0)
                            $('.tmjs-modal').remove();
                        venipak_custom_modal_cod();
                        filterEventListener();
                    }
                    else
                    {
                        document.location = $('.venipakcod').attr('href');
                    }
                },
            });
        }
    });
}

function filterEventListener()
{
    $(".mjvp-pickup-filter").on('click', e => {
        venipak_custom_modal_cod.tmjs.dom.addOverlay();
        const clickTarget = $(e.target);
        if(clickTarget.hasClass('reset'))
        {
            $("#filter-container input[type='checkbox']").each((i, el) => {
                $(el).prop('checked', true);
            });
        }

        var selectedFilters = {};
        var countChecked = 0;
        $("#filter-container input[type='checkbox']").each((i, el) => {
            if($(el).is(':checked'))
            {
                countChecked++;
                selectedFilters['type'] = $(el).data('filter');
            }
        });
        if(countChecked == 2)
            selectedFilters = {};
        else if(countChecked == 0)
            selectedFilters['type'] = 0;
        selectedFilters['cod_enabled'] = 1;

        $('.mjvp-pickup-filter').removeClass('active');
        filterTerminalsAjax(selectedFilters);
    });
}

function filterTerminalsAjax(filters)
{
    $.ajax({
        type: "POST",
        url: mjvp_front_controller_url + "?ajax=1&submitFilterTerminals=1&action=filter",
        dataType: "json",
        data: {
            'filter_keys' : filters
        },
        success: function (res) {
            venipak_custom_modal_cod.tmjs.dom.removeOverlay();
            if(typeof res.mjvp_terminals != "undefined")
            {
                var terminals = [];
                mjvp_terminals = res.mjvp_terminals;
                mjvp_terminals.forEach((terminal) => {
                    if(terminal.lat != 0 && terminal.lng != 0 && terminal.terminal)
                    {
                        terminal['coords'] = {
                            lat: terminal.lat,
                            lng: terminal.lng
                        };
                        terminal['identifier'] = 'venipak';
                        terminals.push(terminal);
                    }
                });
                if(terminals.length == 0)
                {
                    venipak_custom_modal_cod.tmjs.map._markerLayer.clearLayers();
                }
                else
                {
                    venipak_custom_modal_cod.tmjs.setTerminals(terminals);
                    venipak_custom_modal_cod.tmjs.dom.renderTerminalList(venipak_custom_modal_cod.tmjs.map.locations);
                }
            }
        },
    });
}

function mjvp_registerSelectionCod(selected_field_id, ajaxData = {}, params = {}) {
    if(document.getElementById(selected_field_id))
        ajaxData.selected_terminal = document.getElementById(selected_field_id).value;
    if(document.getElementById("mjvp-pickup-country"))
        ajaxData.country_code = document.getElementById("mjvp-pickup-country").value;

    var terminal = null;
    mjvp_terminals.forEach((val, i) => {
        if(parseInt(val.id) == parseInt(ajaxData.selected_terminal)) {
            terminal = val;
        }
    });
    ajaxData.terminal = terminal;

    if(terminal && terminal.cod_enabled == 1)
    {
        // 1.6
        $('.venipak-service-content').remove();
        $('.venipakcod .alert').remove();
        // 1.7
        var venipakCodeElement = $('[data-module-name="venipakcod"]');
        venipakCodeElement.closest('.payment-options').find('.additional-information .alert').remove();
        $('#conditions-to-approve input[type="checkbox"]').removeAttr('disabled');
        $('#payment-confirmation button').removeAttr('disabled');
    }

    $.ajax(mjvp_front_controller_url,
        {
            data: ajaxData,
            type: "POST",
            dataType: "json",
        })
        .always(function (jqXHR, status) {
            if(typeof jqXHR.errors != 'undefined')
            {
                $('[id^="delivery_option"]:checked').parents('.delivery_option ').prepend(jqXHR.errors);
                if(typeof params.scrollToError != "undefined")
                {
                    $([document.documentElement, document.body]).animate({
                        scrollTop: $(".alert.alert-danger").offset().top - 100
                    }, 800);
                }
            }
            else if(typeof params.href != "undefined")
            {
                document.location = params.href;
            }
            if (typeof jqXHR === 'object' && jqXHR !== null && 'msg' in jqXHR) {
                console.log(jqXHR.msg);
            } else {
                console.log(jqXHR);
            }
        });
}

// 1.7
$(document).on('ready', () => {
    var venipakCodeElement = $('[data-module-name="venipakcod"]');
    var additionalInfoElement = venipakCodeElement.closest('.payment-options').find('.additional-information');
    venipakCodeElement.on('click', () => {
        $('.venipak-service-content').remove();
        additionalInfoElement.find('.alert').remove();
        $(".mjvp-pickup-filter").unbind('click');
        $.ajax({
            type: "POST",
            url: cod_ajax_url,
            dataType: "json",
            success: function (res) {
                if(typeof res.carrier_content != 'undefined' && typeof res.mjvp_map_template != 'undefined' && res.carrier_content)
                {
                    var error = '';
                    var tosEl = $('#conditions-to-approve input[type="checkbox"]');
                    tosEl.attr('disabled', true);
                    $('#payment-confirmation button').attr('disabled', true);
                    tosEl.removeAttr('checked');
                    if(typeof res.error != 'undefined')
                    {
                        error = `<div class="alert alert-danger">${res.error}</div>`;
                    }
                    additionalInfoElement.append(error);
                    mjvp_map_template = res.mjvp_map_template;
                    additionalInfoElement.append(`
                        <div class="venipak-service-content">
                            ${res.carrier_content}
                        </div>`);
                    venipak_custom_modal_cod();
                    filterEventListener();
                }
            },
        });
    });

    // When non Venipak COD payment method is selected, we remove the artificial TOS disable and error message.
    $("input[id^='payment-option']:not([data-module-name='venipakcod'])").on('click', () => {
        $('#conditions-to-approve input[type="checkbox"]').removeAttr('disabled');
        $('[data-module-name="venipakcod"]').closest('.payment-options').find('.alert').remove();
    });
})